src/business_logic/state/state_api.rs

```diff
@@ -11,37 +11,19 @@ use cairo_lang_starknet::casm_contract_class::CasmContractClass;
 use cairo_vm::felt::Felt252;
 
 pub trait StateReader {
-    /// Returns the contract class of the given class hash.
-    fn get_contract_class(&mut self, class_hash: &ClassHash) -> Result<ContractClass, StateError>;
+    /// Returns the contract class of the given class hash or compiled class hash.
+    fn get_contract_class(&mut self, class_hash: &ClassHash) -> Result<CompiledClass, StateError>;
     /// Returns the class hash of the contract class at the given address.
     fn get_class_hash_at(&mut self, contract_address: &Address) -> Result<ClassHash, StateError>;
     /// Returns the nonce of the given contract instance.
     fn get_nonce_at(&mut self, contract_address: &Address) -> Result<Felt252, StateError>;
     /// Returns the storage value under the given key in the given contract instance.
     fn get_storage_at(&mut self, storage_entry: &StorageEntry) -> Result<Felt252, StateError>;
-    /// Return de casm contract class of the given class hash.
-    fn get_compiled_class(
-        &mut self,
-        compiled_class_hash: &CompiledClassHash,
-    ) -> Result<CompiledClass, StateError>;
     /// Return the class hash of the given casm contract class
     fn get_compiled_class_hash(
```

Comment:
do we really need get_compiled_class_hash? Maybe we can remove it from the trait and leave it as an implementation detail.

Reply:
We don't need it for the `CachedState`'s public api, but we do need it for the state reader inside it, currently `InMemoryStateReader`. Every time we fetch a contract from the state_reader, we add it to the cache, which is not possible to do if we can't get the compiled class hash from the state